sgl-kernel/csrc/moe/moe_sum_reduce.cu

```diff
@@ -0,0 +1,315 @@
+#include <ATen/cuda/CUDAContext.h>
+#include <c10/cuda/CUDAGuard.h>
+#include <cuda.h>
+#include <cudaTypedefs.h>
+#include <cuda_runtime.h>
+#include <torch/all.h>
+
+#include <iostream>
+#include <type_traits>
+
+#include "cutlass/array.h"
+#include "utils.h"
+
+template <typename T>
```

Comment:
Why not use [`castToFloat` and `castFromFloat`](https://github.com/sgl-project/sglang/blob/main/sgl-kernel/include/utils.h#L348-L357)

Reply:
castToFloat doesn't optimize for half type.
```
template <typename srcDtype>
__device__ __forceinline__ float castToFloat(srcDtype val) {
  return static_cast<srcDtype>(val);
}

```
Prefer to keep the current implementation.